src/basic/mountpoint-util.c

```diff
@@ -602,3 +607,60 @@ int mount_propagation_flag_from_string(const char *name, unsigned long *ret) {
 bool mount_propagation_flag_is_valid(unsigned long flag) {
         return IN_SET(flag, 0, MS_SHARED, MS_PRIVATE, MS_SLAVE);
 }
+
+unsigned long ms_nosymfollow_supported(void) {
+        _cleanup_close_ int fsfd = -EBADF, mntfd = -EBADF;
+        static int cache = -1;
+
+        /* Returns MS_NOSYMFOLLOW if it is supported, zero otherwise. */
+
+        if (cache >= 0)
+                return cache ? MS_NOSYMFOLLOW : 0;
+
+        /* Checks if MS_NOSYMFOLLOW is supported (which was added in 5.10). We use the new mount API's
+         * mount_setattr() call for that, which was added in 5.12, which is close enough. */
+
+        fsfd = fsopen("tmpfs", FSOPEN_CLOEXEC);
+        if (fsfd < 0) {
+                if (ERRNO_IS_NOT_SUPPORTED(errno))
+                        goto not_supported;
+
+                log_debug_errno(errno, "Failed to open superblock context for tmpfs: %m");
+                return 0;
```

Comment:
we usually avoid caching "uncertainties". we only cache reliable results, in the hope that failures might be transient. After all these are unexpected failures, and can be anything, they could be systematic, or transient we simply don't know. policy stuff for example likely is transient (for example: we might boot up with a different MAC policy initially than is applied later, simply because its loaded from userspace).

I think this is a good rule to stick to: cache expected cases, not unexpected ones, we simply can't know if they are transient or persistent failures...

Reply:
OK, that makes sense to me.